src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerForTag.java

```diff
@@ -0,0 +1,76 @@
+package com.hubspot.jinjava.lib.tag.eager;
+
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.interpret.TemplateSyntaxException;
+import com.hubspot.jinjava.lib.tag.ForTag;
+import com.hubspot.jinjava.tree.parse.TagToken;
+import com.hubspot.jinjava.util.ChunkResolver;
+import com.hubspot.jinjava.util.HelperStringTokenizer;
+import java.util.HashSet;
+import java.util.List;
+import java.util.StringJoiner;
+
+public class EagerForTag extends EagerTagDecorator<ForTag> {
+
+  public EagerForTag() {
+    super(new ForTag());
+  }
+
+  public EagerForTag(ForTag forTag) {
+    super(forTag);
+  }
+
+  @Override
+  public String getEagerTagImage(TagToken tagToken, JinjavaInterpreter interpreter) {
+    List<String> helperTokens = new HelperStringTokenizer(
+      ForTag.getWhitespaceAdjustedHelpers(tagToken.getHelpers())
+    )
+      .splitComma(true)
+      .allTokens();
+    List<String> loopVars = getTag().getLoopVars(helperTokens);
+    if (loopVars.size() >= helperTokens.size()) {
+      throw new TemplateSyntaxException(
+        tagToken.getHelpers().trim(),
+        "Tag 'for' expects valid 'in' clause, got: " + tagToken.getHelpers(),
+        tagToken.getLineNumber(),
+        tagToken.getStartPosition()
+      );
+    }
+
+    String loopExpression = getTag().getLoopExpression(helperTokens, loopVars);
+    ChunkResolver chunkResolver = new ChunkResolver(
+      loopExpression,
+      tagToken,
+      interpreter
+    );
+
+    StringJoiner joiner = new StringJoiner(" ");
```

Comment:
There's a guava joiner which I see used more commonly. One advantage is that you can declare it statically. I'm not sure if it's worth changing to that one.

Reply:
I'll be swapping these usages out with [the length limiting joiner that I'm adding](https://github.com/HubSpot/jinjava/blob/3097aa7eabba4378d8331a15dd60a95b78722744/src/main/java/com/hubspot/jinjava/util/LengthLimitingStringJoiner.java) so I will see which makes more sense to use in that class.
Maybe I'll run a simple performance test to see which is faster

Reply:
Looks like the java.util.StringJoiner is 2-3x faster than the guava joiner so I'll stick with it.
Time to join 10k and 1 million `a` characters with a `,` character:

| Number of elements |StringJoiner | Guava Joiner |
| --- | --- | --- |
| 10000 | 1 ms | 8 ms |
| 1000000 | 17 ms | 51 ms |

Reply:
Interesting. Thanks for verifying. The JDK StringJoiner was introduced in Java 8, so maybe it's just historical that we use it. That said, the guava version has some nice features like `skipNulls`